File: day02/d2p1.py
def winDraw(line):
    if (line[0]=="A" and line[2]=="Y") or (line[0]=="B" and line[2]=="Z") or (line[0]=="C" and line[2]=="X"):
        return 6
    if (line[0]=="A" and line[2]=="X") or (line[0]=="B" and line[2]=="Y") or (line[0]=="C" and line[2]=="Z"):
        return 3
    return 0

count = 0
val = {"X": 1, "Y": 2, "Z": 3}
with open('input.txt', 'r') as f:
    for line in f:
        count += val[line[2]]
        count += winDraw(line)
    
    print(count)

File: day02/d2p1.hs
score :: String -> Int
score "A X" = 4
score "A Y" = 8
score "A Z" = 3
score "B X" = 1
score "B Y" = 5
score "B Z" = 9
score "C X" = 7
score "C Y" = 2
score "C Z" = 6

read_numbers = do
    contents <- readFile "input.txt"
    -- let linesOfFile = map (\x -> read x :: String) (lines $ contents)
    let linesOfFile = (lines $ contents)
    return linesOfFile

main = do
      xs <- read_numbers
      print (sum (map score xs))